app/views/idv/doc_auth/verify.html.erb

```diff
@@ -57,25 +57,29 @@
       </div>
     </div>
   <div class="margin-top-5">
-    <%= render 'shared/spinner_button',
-               action_message: t('doc_auth.info.verifying'),
-               class: 'grid-col-12 tablet:grid-col-6' do %>
-      <%= button_to(
-            url_for,
-            method: :put,
-            form: {
-              class: 'button_to',
-              data: {
-                form_steps_wait: '',
-                error_message: t('idv.failure.exceptions.internal_error'),
-                alert_target: '#form-steps-wait-alert',
-                wait_step_path: idv_doc_auth_step_path(step: :verify_wait),
-                poll_interval_ms: IdentityConfig.store.poll_rate_for_verify_in_seconds * 1000,
+    <%= render SpinnerButtonComponent.new(
+          action: ->(**tag_options, &block) do
+            button_to(
+              url_for,
+              **tag_options,
+              method: :put,
+              form: {
+                class: 'button_to',
+                data: {
+                  form_steps_wait: '',
+                  error_message: t('idv.failure.exceptions.internal_error'),
+                  alert_target: '#form-steps-wait-alert',
+                  wait_step_path: idv_doc_auth_step_path(step: :verify_wait),
+                  poll_interval_ms: IdentityConfig.store.poll_rate_for_verify_in_seconds * 1000,
+                },
```

Comment:
this seems like a really complicated usage of our `action` callback... I feel like most of those `data-` attrbiutes belong on the outside form, would it make sense to have the `SpinnerButtonComponent` take in any of these and pass them in to the Button action callback as `tag_options`?

Reply:
Ah yeah, that's a good point. Any extra attributes we assign to the component constructor should get passed through automatically, so we can add them there.

Generally I'm still not feeling 100% about `action` in general, though this particular button is also made much more complex with how it's initializing the `form-steps-wait` behavior, so not entirely the fault of the button API.